### PR TITLE
Miner testing

### PR DESCRIPTION
**Pull request by ortyomka**
Test for config in store

Signed-off-by: ortyomka <iurin.art@gmail.com>

### CODE REVIEW

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  /**
+   * @brief Instance of cursor can be used as bidirectional iterator over
+   * key-value bindings of the Map.
+   */
+  class InMemoryCursor : public BufferMapCursor {
+   public:
+    InMemoryCursor(std::shared_ptr<InMemoryStorage> storage)
+        : storage_(std::move(storage)) {
+      current_iterator_ = storage_->storage.end();
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:20]:
```suggestion
      current_iterator_ = invalid_;
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    };
+
+    void seekToFirst() override {
+      current_iterator_ = storage_->storage.begin();
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:24]:
```suggestion
      current_iterator_ = storage_->storage.begin();
      if (current_iterator_ == storage_->storage.end()) {
        current_iterator_ = invalid_;
      }
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    void seek(const Buffer &key) override {
+      current_iterator_ = storage_->storage.find(key.toHex());
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:28]:
```suggestion
      current_iterator_ = storage_->storage.lower_bound(key.toHex());
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    void seekToLast() override {
+      current_iterator_ = (storage_->storage.rbegin()++).base();
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:32]:
```suggestion
      current_iterator_ = storage_->storage.end();
      if (current_iterator_ == storage_->storage.begin()) {
        current_iterator_ = invalid_;
      } else {
        --current_iterator_;
      }
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    bool isValid() const override {
+      return current_iterator_ != storage_->storage.end();
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:36]:
```suggestion
      return current_iterator_ != invalid_;
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    void next() override {
+      current_iterator_++;
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:40]:
```suggestion
      assert(isValid());
      current_iterator_++;
      if (current_iterator_ == storage_->storage.end()) {
        current_iterator_ = invalid_;
      }
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    void prev() override {
+      current_iterator_--;
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:44]:
```suggestion
      assert(isValid());
      if (current_iterator_ == storage_->storage.begin()) {
        current_iterator_ = invalid_;
      } else {
        current_iterator_--;
      }
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    Buffer key() const override {
+      OUTCOME_EXCEPT(key, Buffer::fromHex(current_iterator_->first));
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:48]:
```suggestion
      return Buffer::fromHex(current_iterator_->first).value();
```

File: core/storage/in_memory/in_memory_cursor.hpp [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      return key;
+    }
+
+    Buffer value() const override {
+      return current_iterator_->second;
+    }
+
+   private:
+    std::map<std::string, Buffer>::iterator current_iterator_;
+
```
Comment (turuslan) [inline core/storage/in_memory/in_memory_cursor.hpp:58]:
```suggestion
    std::map<std::string, Buffer>::iterator current_iterator_, invalid_;

```

File: core/sector_storage/stores/impl/util.hpp [hunk 10]
```diff
@@ -21,7 +21,7 @@ namespace fc::sector_storage::stores {
       fs::create_directories(temp_dir, ec);
       if (ec.failed()) {
         // TODO: log error
-        return StoreErrors::kCannotCreateDir;
+        return StoreError::kCannotCreateDir;
```
Comment (Alexey-N-Chernyshov) [inline core/sector_storage/stores/impl/util.hpp:24]: Please, add log or remove TODO.

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Comment by Alexey-N-Chernyshov** [comment]
Do not forget enable ubsan and lsan.

File: core/storage/in_memory/in_memory_cursor.hpp (hunk 1 above)
Comment (ortyomka) [reply core/storage/in_memory/in_memory_cursor.hpp]: Fixed

File: core/storage/in_memory/in_memory_cursor.hpp (hunk 2 above)
Comment (ortyomka) [reply core/storage/in_memory/in_memory_cursor.hpp:24]: Fixed

File: core/storage/in_memory/in_memory_cursor.hpp (hunk 3 above)
Comment (ortyomka) [reply core/storage/in_memory/in_memory_cursor.hpp]: Fixed

File: core/storage/in_memory/in_memory_cursor.hpp (hunk 5 above)
Comment (ortyomka) [reply core/storage/in_memory/in_memory_cursor.hpp]: Fixed

File: core/storage/in_memory/in_memory_cursor.hpp (hunk 7 above)
Comment (ortyomka) [reply core/storage/in_memory/in_memory_cursor.hpp]: Fixed

File: core/storage/in_memory/in_memory_cursor.hpp (hunk 9 above)
Comment (ortyomka) [reply core/storage/in_memory/in_memory_cursor.hpp]: Fixed

File: core/sector_storage/stores/impl/util.hpp (hunk 10 above)
Comment (ortyomka) [reply core/sector_storage/stores/impl/util.hpp:27]: Added

**Review by turuslan — APPROVED** [review]

**Pull request merged**